src/agent.rs

```diff
@@ -16,25 +19,43 @@ pub struct Agent {
     path: String,
     args: Vec<String>,
     pub socket: TcpStream,
-    child: Receiver<i32>,
+    child: Receiver<Output>,
     pub alive: bool,
     exit_value: Option<ExitStatus>,
 }
 
+fn cipher_string_to_ossl(input: &str) -> String {
+    String::from(input)
+        .replace("TLS_", "")
+        .replace("AES_", "AES")
+        .replace("_", "-")
+        .replace("-WITH", "")
+}
 
 impl Agent {
-    pub fn new(name: &str,
-               path: &String,
-               agent: &Option<TestCaseAgent>,
-               args: Vec<String>)
-               -> Result<Agent, i32> {
-        let addr = "127.0.0.1:0".parse().unwrap();
-        let listener = TcpListener::bind(&addr).unwrap();
-
+    pub fn new(
+        name: &str,
+        path: &str,
+        agent: &Option<TestCaseAgent>,
+        args: Vec<String>,
+        ipv4: bool,
+    ) -> Result<Agent, i32> {
+        // IPv6 listener by default, IPv4 fallback, unless IPv4 is forced.
+        let addr6 = "[::1]:0".parse().unwrap();
+        let addr4 = "127.0.0.1:0".parse().unwrap();
```

Comment:
I'm not sure I understand what you mean. I've personally found this the most readable and compact way to achieve what I described in the comment. The reason to introduce the flag that forces IPv4 was that the OpenSSL-Shim is not IPv6-compatible yet, even in an environment that offers IPv6 and thus lets interop open an IPv6 socket.

Reply:
What I'm saying is put the defn of |addr4| and |addr6| inside the appropriate arms of the match.

Reply:
Mmm, since `addr4` is needed in both branches of the match, I'd vote for keeping it as Jonas has it. I think we'd need to change this to a little for loop to have single declarations and accomplish what we're looking for.